### DISCUSSION TITLE
allow to clean Notifications.useLastNotificationResponse() once handled

### DISCUSSION THREAD

**Discussion by wcastand** [expo/expo#22302] [Proposals and Ideas]
using expo-router and figured i needed to use Notifications.useLastNotificationResponse() because when the app is killed, i don't get the notification listener to trigger which makes it not possible to handle the deeplink in it.

`Notifications.useLastNotificationResponse()` is a possible solution to handle all the use case (background, killed) but the issue is that once the notif is handled i can't find a way to clear `Notifications.useLastNotificationResponse()` data, so everytime i check the app think there is a deeplink to open.

Happens especially in dev when you test one notification and then make a change in code and reload happens, it will re-handle the deeplink and i can't find a way to prevent that.

Notification listeners don't handle the case of the app being killed and the user tap on a notification with a deeplink. (hard to actually test in dev since the tap opens the expo-dev-app which ask to select a server and loose the notification data)

Is there something i missed in the different use case or is it possible to clear `Notifications.useLastNotificationResponse()` ?

**Comment by Odyrac** [comment]
up
